File: src/main/java/gg/manny/forums/user/punishment/PunishmentType.java
package gg.manny.forums.user.punishment;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum PunishmentType {

    WARN("warned", "#fff"),
    KICK("kicked", "#E74C3C"),
    MUTE("muted", "#F8C471"),
    BAN("banned", "#B03A2E"),
    BLACKLIST("blacklisted", "#B03A2E");

    private final String action;
    private final String color;

}

File: src/main/kotlin/dev/foraged/forums/user/punishment/PunishmentType.kt
package dev.foraged.forums.user.punishment

enum class PunishmentType(val action: String, val color: String)
{
    WARN("warned", "#fff"),
    MUTE("muted", "#F8C471"),
    KICK("kicked", "#E74C3C"),
    BAN("banned", "#B03A2E"),
    BLACKLIST("blacklisted", "#B03A2E");
}